naturblick/utils/ObservationPersistenceController.swift

```diff
@@ -317,22 +317,21 @@ class ObservationPersistenceController: ObservableObject {
         }
     }
     
-    func delete(indexSet: IndexSet) throws {
-        try queue.transaction {
-            for i in indexSet {
-                let deleteId = try queue.run(Operation.D.table.insert())
-                try queue.run(DeleteOperation.D.table.insert(DeleteOperation(occurenceId: observations[i].id).setters(id: deleteId)))
-            }
-            try updateAndRefresh()
-        }
-    }
-    
     func delete(occurenceId: UUID) throws {
         let delete = DeleteOperation(occurenceId: occurenceId)
         let deleteId = try queue.run(Operation.D.table.insert())
         try queue.run(DeleteOperation.D.table.insert(delete.setters(id: deleteId)))
         try updateAndRefresh()
     }
+    
+    func delete(occurenceIds: [UUID]) throws {
+        for occurenceId in occurenceIds {
```

Comment:
I think it makes sense to do this in a transaction, otherwise it would be hard to reason about what happens when another thread (like server sync) updates the DB at the same time.